### PR TITLE
WRQ-8002: Applied UX scenario to TransferList sample

### PR DESCRIPTION
**Pull request by ion-andrusciac-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Applied UX scenario to TransferList sample

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added remove button, fixed Drop Border, and added the option of fixing the items from the first list

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-8002

### Comments

Enact-DCO-1.0-Signed-off-by: Ion Andrusciac ion.andrusciac@lgepartner.com

### CODE REVIEW

File: TransferList/ButtonList.js [hunk 1]
```diff
@@ -80,6 +92,14 @@ ButtonList.propTypes = {
 	 */
 	disabled: PropTypes.bool,
 
+	/**
+	 * Disables Remove Button when items are not selected or dragged
+	 *
+	 * @type {Boolean}
+	 * @private
+	 */
+	elementsAreDragged: PropTypes.bool,
```
Comment (adrian-cocoara-lgp) [inline TransferList/ButtonList.js:101]: if this prop only applies to `Remove button` we need to find a better name. maybe `trashBinActive`? any suggetions are welcomed

File: TransferList/TransferList.js [hunk 2]
```diff
@@ -96,6 +96,15 @@ const TransferListBase = kind({
 		 */
 		firstListOperation: PropTypes.oneOf(['move', 'copy', 'delete']),
 
+		/**
+		 * Fixes the order of items on the first list.
```
Comment (adrian-cocoara-lgp) [inline TransferList/TransferList.js:100]:
```suggestion
		 * Fixes the order of items on the first list. The items cannot be re-arranged or deleted
```

**Review by daniel-stoian-lgp — APPROVED** [review]

File: TransferList/TransferList.js [hunk 3]
```diff
@@ -238,7 +248,8 @@ const TransferListBase = kind({
 		publicClassNames: true
 	},
 
-	render: ({css, disabled, firstList, firstListMaxCapacity, firstListMinCapacity, firstListOperation, height: defaultHeight, itemSize: defaultItemSize, listComponent, moveOnSpotlight, noMultipleSelect, orientation, secondList, secondListMaxCapacity, secondListMinCapacity, secondListOperation, setFirstList, setSecondList, showSelectionOrder, verticalHeight}) => {
+	render: ({css, disabled, firstList, firstListOrderFixed, firstListMaxCapacity, firstListMinCapacity, firstListOperation, height: defaultHeight, itemSize: defaultItemSize, listComponent, moveOnSpotlight, noMultipleSelect, orientation, secondList, secondListMaxCapacity, secondListMinCapacity, secondListOperation, setFirstList, setSecondList, showSelectionOrder, verticalHeight}) => {
+		const [elementsAreDragged, setElementsAreDragged] = useState(false);
```
Comment (adrian-cocoara-lgp) [inline TransferList/TransferList.js:252]: could you leave a comment here that explains what this state represents?

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM 👍

**Pull request merged**